### PR TITLE
Add paginated artifact list

### PR DESCRIPTION
**Pull request by found-it**
Now that we are generating more artifacts in the release, they won't all fit into the default page size. This just adds a function to search each page for an artifact before retrieving the next page.

### CODE REVIEW

File: modules/github-bots/sdk/github.go [hunk 1]
```diff
@@ -83,6 +85,60 @@ func (c GitHubClient) Close(ctx context.Context) error {
...
+// returned by the API.
+func handleGithubResponse(ctx context.Context, resp *github.Response, err error) error {
+	// if err is rate limit, delay and try again
+	githubErr := github.CheckResponse(resp.Response)
+	if githubErr != nil {
+		rateLimited, delay := checkRateLimiting(ctx, githubErr)
+		if !rateLimited {
+			return err
+		}
+		fmt.Printf("retrying after %v second delay due to rate limiting\n", delay.Seconds())
```
Comment (imjasonh) [inline modules/github-bots/sdk/github.go:133]: Use clog here

File: modules/github-bots/sdk/github.go [hunk 2]
```diff
@@ -83,6 +85,60 @@ func (c GitHubClient) Close(ctx context.Context) error {
...
+	isRateLimited := false
+
+	// If GitHub returned an error of type RateLimitError, we can attempt to compute the next time to try the request again
+	// by reading its rate limit information
+	var rateLimitError *github.RateLimitError
+	if errors.As(githubErr, &rateLimitError) {
+		isRateLimited = true
+		retryAfter := time.Until(rateLimitError.Rate.Reset.Time)
+		delay = retryAfter
+		log.Infof("parsed retryAfter %d from GitHub rate limit error's reset time", retryAfter)
```
Comment (imjasonh) [inline modules/github-bots/sdk/github.go:106]: This might be spammy, maybe Debugf?

File: modules/github-bots/sdk/github.go [hunk 3]
```diff
@@ -83,6 +85,60 @@ func (c GitHubClient) Close(ctx context.Context) error {
...
+func handleGithubResponse(ctx context.Context, resp *github.Response, err error) error {
+	// if err is rate limit, delay and try again
+	githubErr := github.CheckResponse(resp.Response)
+	if githubErr != nil {
+		rateLimited, delay := checkRateLimiting(ctx, githubErr)
+		if !rateLimited {
+			return err
+		}
+		fmt.Printf("retrying after %v second delay due to rate limiting\n", delay.Seconds())
+		time.Sleep(delay)
```
Comment (imjasonh) [inline modules/github-bots/sdk/github.go:134]: This makes me a little nervous tbh.

GH's rate limit quota resets hourly, so if we blow through it all at once then we could have jobs sleeping for 50m for the reset, costing money while they wait to time out and retry. Multiply that by potentially dozens of replicas, and it's not great.

I think we need a different approach to rate limits in the bot world, and we have some ideas.

Comment (found-it) [reply modules/github-bots/sdk/github.go:134]: Good note @imjasonh

I don't need the rate limiting to complete my cycle item. Want me to just remove it from this PR and we can chat about the best way to do it moving forward?

File: modules/github-bots/sdk/github.go (hunk 1 above)
Comment (found-it) [reply modules/github-bots/sdk/github.go:133]: Good catch!

File: modules/github-bots/sdk/github.go (hunk 3 above)
Comment (joshrwolf) [reply modules/github-bots/sdk/github.go:134]: +1, lets separate the contentious rate-limiting parts and land this

Comment (found-it) [reply modules/github-bots/sdk/github.go]: separated ✅

**Review by imjasonh — APPROVED** [review]

**Pull request merged**